v0/destinations/blueshift/transform.js

```diff
@@ -0,0 +1,212 @@
+      400
+    );
+  }
+
+  if (!Config.eventApiKey) {
+    throw new CustomError(
+      "[BLUESHIFT] event Api Keys required for Authentication.",
+      400
+    );
+  }
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+
+  event = event.trim();
+  if (isDefinedAndNotNull(EVENT_NAME_MAPPING[event])) {
+    payload.event = EVENT_NAME_MAPPING[event];
+  }
+  payload.event = payload.event.replace(/\s+/g, "_");
+  if (checkValidEventName(payload.event)) {
+    throw new CustomError(
+      "[Blueshift] Event shouldn't contain period(.), numeric value and contains not more than 64 characters",
+      400
+    );
+  }
+  const response = defaultRequestConfig();
+
+  if (Config.datacenterEU) {
+    response.endpoint = `${BASE_URL_EU}/api/v1/event`;
+  } else {
+    response.endpoint = `${BASE_URL}/api/v1/event`;
+  }
+  response.method = defaultPostRequestConfig.requestMethod;
+  const basicAuth = Buffer.from(Config.eventApiKey).toString("base64");
+  response.headers = {
+    Authorization: `Basic ${basicAuth}`,
+    "Content-Type": "application/json"
+  };
+  response.body.JSON = payload;
+  return response;
+};
+
+const identifyResponseBuilder = async (message, category, { Config }) => {
+  if (!Config.usersApiKey) {
+    throw new CustomError(
+      "[BLUESHIFT] User API Key required for Authentication.",
+      400
+    );
+  }
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+  const response = defaultRequestConfig();
+
+  if (Config.datacenterEU) {
+    response.endpoint = `${BASE_URL_EU}/api/v1/customers`;
+  } else {
+    response.endpoint = `${BASE_URL}/api/v1/customers`;
+  }
+  response.method = defaultPostRequestConfig.requestMethod;
+  const basicAuth = Buffer.from(Config.usersApiKey).toString("base64");
+  response.headers = {
+    Authorization: `Basic ${basicAuth}`,
+    "Content-Type": "application/json"
+  };
+  response.body.JSON = payload;
+  return response;
+};
+
+const groupResponseBuilder = async (message, category, { Config }) => {
+  if (!Config.usersApiKey) {
+    throw new CustomError(
+      "[BLUESHIFT] User API Key required for Authentication.",
+      400
+    );
+  }
+
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+  if (payload.name && payload.description) {
+    let endpoint;
+    if (Config.datacenterEU) {
+      endpoint = `${BASE_URL_EU}/api/v1/custom_user_lists/create`;
+    } else {
+      endpoint = `${BASE_URL}/api/v1/custom_user_lists/create`;
+    }
```

Comment:
You can simplifiy the logic
```suggestion
    let baseURL = Config.datacenterEU ? BASE_URL_EU : BASE_URL;
    let endpoint = `${baseURL}/api/v1/custom_user_lists/create`;
```